realm/realm-library/src/androidTest/kotlin/io/realm/DictionaryTests.kt

```diff
@@ -196,44 +231,312 @@ class DictionaryTests {
+//            val dictionaryFromRealm = dictionaryObjectFromRealm.myStringDictionary
+//            assertNotNull(dictionaryFromRealm)
+//
+//            assertEquals(VALUE_HELLO_STRING, dictionaryFromRealm[KEY_HELLO])
+//            assertEquals(VALUE_BYE_STRING, dictionaryFromRealm[KEY_BYE])
+//            assertNull(dictionaryFromRealm[KEY_NULL])
+//        }
+//    }
+
+//    @Test
+//    fun copyToRealm_integer() {
+//        realm.executeTransaction { transactionRealm ->
+//            val dictionaryObject = DictionaryClass().apply {
+//                myIntegerDictionary = RealmDictionary<Int>().apply {
+//                    put(KEY_HELLO, VALUE_HELLO_INTEGER)
+//                    put(KEY_BYE, VALUE_BYE_INTEGER)
+//                    put(KEY_NULL, null)
+//                }
+//            }
+//
+//            val dictionaryObjectFromRealm = transactionRealm.copyToRealm(dictionaryObject)
+//            val dictionaryFromRealm = dictionaryObjectFromRealm.myIntegerDictionary
+//            assertNotNull(dictionaryFromRealm)
+//
+//            assertEquals(VALUE_HELLO_INTEGER, dictionaryFromRealm[KEY_HELLO])
+//            assertEquals(VALUE_BYE_INTEGER, dictionaryFromRealm[KEY_BYE])
+//            assertNull(dictionaryFromRealm[KEY_NULL])
+//        }
+//    }
+
+    private fun createBooleanRealmDictionary(
+            withNullValues: Boolean = false
+    ): RealmDictionary<Boolean> {
+        return RealmDictionary<Boolean>().apply {
+            put(KEY_HELLO, VALUE_HELLO)
+            put(KEY_BYE, VALUE_BYE)
+
+            if (withNullValues) {
+                put(KEY_NULL, VALUE_NULL)
+            }
+        }
+    }
+
+    private fun createMixedRealmDictionary(): RealmDictionary<Mixed> {
+        return RealmDictionary<Mixed>().apply {
+            put(KEY_HELLO, Mixed.valueOf(VALUE_HELLO))
+            put(KEY_BYE, Mixed.valueOf(VALUE_BYE))
+        }
+    }
+
+    private fun initDictionary() {
+        realm.executeTransaction { transactionRealm ->
+            transactionRealm.createObject<DictionaryClass>()
+                    .also { it.myBooleanDictionary = createBooleanRealmDictionary() }
+        }
+    }
+
+    private fun baseAssertions(dictionary: RealmDictionary<*>?) {
+        assertNotNull(dictionary)
+        assertFalse(dictionary.isEmpty())
+        assertEquals(2, dictionary.size)
+        assertEquals(VALUE_HELLO, dictionary[KEY_HELLO])
+        assertEquals(VALUE_BYE, dictionary[KEY_BYE])
+    }
+
+    private fun initDictionaryAndAssert(): RealmDictionary<*> {
+        initDictionary()
+
+        val dictionaryObject = realm.where<DictionaryClass>().findFirst()
+        assertNotNull(dictionaryObject)
+
+        return dictionaryObject.myBooleanDictionary!!
+                .also { baseAssertions(it) }
     }
 }
 
-@RealmModule(classes = [DictionaryClass::class, Dog::class, Cat::class, Owner::class, DogPrimaryKey::class])
+@RealmModule(classes = [DictionaryClass::class, MyDog::class])
```

Comment:
Instead of using a `RealmModule` to define the available realm models, you can use the `schema` function in `RealmConfiguration` builder.

Reply:
I'm planning on revamping the tests in another PR. I will consider this then.